software/speller/data_collection_platform/dcp/api.py

```diff
@@ -38,38 +39,128 @@ def index():
     # TODO return keyboard layout
     return ' '
 
-
-@bp.route('/char_pressed', methods=['POST', 'GET'])
-def char_pressed():
-    if request.method == 'POST':
-        user_char_pressed = request.form['char_pressed']
-
-
 @bp.route("/openbci/start", methods=['POST'])
 def openbci_start():
     # use a separate process to stream BCI data
     from dcp.bci.stream import stream_bci_api
     from multiprocessing import Process
     with current_app.app_context():
-        p = Process(target=stream_bci_api, args=(shared.bci_processes_states,))
+        # Although the subprocess gets initialized inside this scope, it will persist
+        # bci_processes_states references this subprocess_dict
+        subprocess_dict = shared.manager.dict({
+            'character': None,
+            'phase': None,
+            'frequency': None,
+            'q': deque(),
+            'config_id': None, 
+            'bci_config': None,
+            'state': 'start'
+        })
```

Comment:
```suggestion
        subprocess_dict = dict({
            'character': None,
            'phase': None,
            'frequency': None,
            'q': shared.manager.Queue(),
            'config_id': None, 
            'bci_config': None,
            'state': None
        })
```

Reply:
Reason I am initializing a normal dict instead of a `shared.manager.dict()` is, because of this is the only way we can wrap shared variables.

Reply:
![image](https://user-images.githubusercontent.com/43357040/149646317-9acaa744-086d-4625-ab10-d7a4b21e87a2.png)